tests/lib/nrf_modem_lib/nrf_modem_lib_trace/prj.conf

```diff
@@ -5,6 +5,7 @@
 #
 
 CONFIG_UNITY=y
-CONFIG_ASSERT=y
-CONFIG_NRF_MODEM_LIB_TRACE_ENABLED=y
-CONFIG_NRF_MODEM_LIB_TRACE_THREAD_PROCESSING=y
+CONFIG_ASSERT=n
+CONFIG_NRF_MODEM_LIB_TRACE=y
```

Comment:
Either add `CONFIG_NRF_MODEM_LIB_TRACE_LEVEL_OVERRIDE=n` or rewrite tests to expect setting the trace level at initialization.